eventhubs/client/README.md

```diff
@@ -62,14 +62,46 @@ EventHubClient client = builder.build();
+
+- A **partition** is an ordered sequence of events that is held in an Event Hub. Partitions are a means of data organization associated with the parallelism required by event consumers.  Azure Event Hubs provides message streaming through a partitioned consumer pattern in which each consumer only reads a specific subset, or partition, of the message stream. As newer events arrive, they are added to the end of this sequence. The number of partitions is specified at the time an Event Hub is created and cannot be changed.
+
+- A **consumer group** is a view of an entire Event Hub. Consumer groups enable multiple consuming applications to each have a separate view of the event stream, and to read the stream independently at their own pace and from their own position.  There can be at most 5 concurrent readers on a partition per consumer group; however it is recommended that there is only one active consumer for a given partition and consumer group pairing. Each active reader receives all of the events from its partition; if there are multiple readers on the same partition, then they will receive duplicate events. 
+
+For more concepts and deeper discussion, see: [Event Hubs Features](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features). Also, the concepts for AMQP are well documented in [OASIS Advanced Messaging Queuing Protocol (AMQP) Version
 1.0](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-overview-v1.0-os.html).
 
 ## Examples
 
 ## Troubleshooting
 
+### Common exceptions
+#### AMQP Exception
+This is a general exception for AMQP related failures, which includes the AMQP errors as ErrorCondition and the context that caused this exception as ErrorContext. 'isTransien' is A boolean indicating if the exception is a transient error or not. If true, then the request
+can be retried; otherwise not.
+
+- ErrorCondition: it contains constants common to the AMQP protocol and constants shared by Azure services. More detail can be found in the link: [AMQP 1.0: Transport Errors](https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-amqp-error) and [Azure Messaging Exceptions](http://go.microsoft.com/fwlink/?LinkId=761101).
+- ErrorContext, it provides context that caused the AmqpException. The erorr occurs could be from AmqpConnection, AmqpSession, or AmqpLink. Such as SessionErrorContext and LinkErrorContext, the context for an error that occurs in an AMQP session and AMQP link, respectively.
+
+#### Operation Cancelled Exception
+It occurs when the underlying AMQP layer encounters an abnormal link abort or the connection is disconnected in an unexpected fashion.
+
+### Other exceptions
+For detailed information about these and other exceptions that may occur, please refer to [Event Hubs messaging exceptions](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions).
```

Comment:
IIRC: https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions is the forward link http://go.microsoft.com/fwlink/?LinkId=761101. Should consolidate them at the bottom of the links section.

Reply:
The forward link forward to Service Bus Messaging Exceptions. Is this a correct link to use?

Reply:
No. Use the Event Hubs one. And replace instances of this fwdlink in the code too.